integration-tests/framework/v1alpha1/framework.go

```diff
@@ -0,0 +1,187 @@
+package v1alpha1
+
+import (
+	"os"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openebs/maya/integration-tests/artifacts"
+	installer "github.com/openebs/maya/integration-tests/artifacts/installer/v1alpha1"
+	"github.com/openebs/maya/integration-tests/kubernetes"
+	"github.com/openebs/maya/pkg/client/k8s/v1alpha1"
+	pod "github.com/openebs/maya/pkg/kubernetes/pod/v1alpha1"
+
+	validatehook "github.com/openebs/maya/pkg/kubernetes/webhook/validate/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	// auth plugins
+	//_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+)
+
+const (
+	// DefaultTimeOut is the default time in seconds
+	// for Eventually block
+	DefaultTimeOut int = 600
+	// DefaultPollingInterval is the default polling
+	// time in seconds for the Eventually block
+	DefaultPollingInterval int = 10
+	// NodeCountOne is the 1 node
+	// needed to run this test
+	NodeCountOne int = 1
+)
+
+var (
+	defaultOpenebsComponents []*installer.DefaultInstaller
+)
+
+// Framework supports common operations used by tests
+type Framework struct {
+	BaseName string
+	// configuration for framework's options
+	Options FrameworkOptions
```

Comment:
In the next frame work PR should have struct
```
type kind string
type FrameworkInstaller struct {
          installer.DefaultInstaller
          unstructObj *unstructured.Unstructured
          artifact artifacts.ArtifacrSourcr // Is it required?
}
type Framework {
	BaseName string
	// configuration for framework's options
	Options FrameworkOptions
        ResourceKind map[kind]FrameworkInstaller // If we have this objects then framework will be aware of what it is installing
}
```
cc: @AmitKumarDas @prateekpandey14